Test/Integration-Test/test_Piece.py

```diff
@@ -0,0 +1,59 @@
+from unittest import TestCase, mock
+
+import Piece
+from Piece import Knight
+from enums import Player
+
+
+class TestKnight(TestCase):
+    def test_get_valid_piece_moves ( self ):
+        game_state = mock.Mock()
+        game_state.board = [[Player.EMPTY for i in range(8)] for j in range(8)]
+        game_state.get_piece = lambda x, y: game_state.board[x][y] if 0 <= x < 8 and 0 <= y < 8 else None
+        game_state.is_valid_piece = lambda x, y: game_state.get_piece(x, y) not in [None, Player.EMPTY]
+        # test case 1: knight in the middle of empty board
+        knight = Piece.Knight('n', 3, 4, 'white')
+        game_state.board[3][4] = knight
+        moves = knight.get_valid_piece_moves(game_state)
+        write_moves = [(1, 3), (1, 5), (2, 2), (2, 6), (4, 2), (4, 6), (5, 5), (5, 3)]
+        self.assertEqual(len(write_moves), len(moves))
+        for move in moves:
+            self.assertIn(move, write_moves)
+
+        # test case 2: knight in the corner of the board with enemy pieces to capture
```

Comment:
In unitesting you should separate each test to different function, it will make the code and the testing itself less messy